libraries/DataFlash/DataFlash_Backend.cpp

```diff
@@ -260,6 +260,31 @@ bool DataFlash_Backend::Log_Write(const uint8_t msg_type, va_list arg_list, bool
     return WritePrioritisedBlock(buffer, msg_len, is_critical);
 }
 
+bool DataFlash_Backend::StartNewLogOK() const
+{
+    if (logging_started()) {
+        return false;
+    }
+    if (_front._log_bitmask == 0) {
+        return false;
+    }
+    if (_front.in_log_download()) {
+        return false;
+    }
+    if (!DataFlash_Backend::WritesOK()) {
```

Comment:
Humm, makes sense. Although I have to say I dislike that the same method have two slightly different meanings whether it is in base class or in derived class. I wonder if the additional checks in the derived `WritesOK` shouldn't be a separate `CanWrite` method (or a better name). Then in `WritePrioritisedBlock` we could do:

```
if(!WritesOK()) { // we shouldn't write
  return false;
}

if(StartNewLogOK()) { // we need to start a new log
  start_new_log();
}

if(!CanWrite()) { // we should be logging but there was a problem and we can't write
  return false;
}
```

Maybe I'm nitpicking here, but I'm not very comfortable with the way it is now.